### PR TITLE
Add skill tab canvas menu options

### PR DESCRIPTION
**Pull request by Nightfirecat**
Skill tab canvas add/remove menu options

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/OverlayManager.java [hunk 1]
```diff
@@ -217,6 +217,17 @@ public synchronized boolean removeIf(Predicate<Overlay> filter)
 		return removeIf;
 	}
 
+	/**
+	 * Returns whether an overlay exists which matches the given predicate.
+	 *
+	 * @param filter Filter predicate function
+	 * @return {@code true} if any overlays match the given filter, {@code false} otherwise
+	 */
+	public synchronized boolean hasOverlay(Predicate<Overlay> filter)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/OverlayManager.java]: This should be named .find or anyMatch (rest of methods in this class already follow the naming in java collections)

File: runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerConfig.java [hunk 2]
```diff
@@ -104,7 +115,7 @@ default OnScreenDisplayMode onScreenDisplayMode()
 	}
 
 	@ConfigItem(
-		position = 4,
+		position = 5,
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerConfig.java]:
```suggestion
		position = 6,
```

File: runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerPlugin.java [hunk 3]
```diff
@@ -375,6 +386,60 @@ public void onGameTick(GameTick event)
...
+		if (!xpTrackerConfig.skillTabOverlayMenuOptions())
+		{
+			return;
+		}
+
+		int widgetIndex = event.getActionParam0();
+		int widgetID = event.getActionParam1();
+		MenuEntry[] menuEntries = client.getMenuEntries();
+
+		if (WidgetInfo.TO_GROUP(widgetID) != WidgetID.SKILLS_GROUP_ID || !event.getOption().startsWith("View"))
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerPlugin.java]:
```suggestion
		if (WidgetInfo.TO_GROUP(widgetID) != WidgetID.SKILLS_GROUP_ID || !event.getOption().startsWith("View") || !xpTrackerConfig.skillTabOverlayMenuOptions())
```

File: runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerPlugin.java [hunk 4]
```diff
@@ -375,6 +386,60 @@ public void onGameTick(GameTick event)
...
+
+	@Subscribe
+	public void onMenuOptionClicked(MenuOptionClicked event)
+	{
+		if (event.getMenuAction().getId() != MenuAction.RUNELITE.getId())
+		{
+			return;
+		}
+
+		final Skill skill = Skill.valueOf(Text.removeTags(event.getMenuTarget()).toUpperCase());
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerPlugin.java]: I don't think this is guaranteed to be a skill?

File: runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerPlugin.java (hunk 3 above)
Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerPlugin.java]: This is already guarded at the top of the function. These guard checks happen after we get the widget ID from the menu entry. I'll shuffle the variable declarations around to make more sense here.

File: runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerPlugin.java [hunk 5]
```diff
@@ -375,6 +386,61 @@ public void onGameTick(GameTick event)
...
+
+	@Subscribe
+	public void onMenuOptionClicked(MenuOptionClicked event)
+	{
+		if (event.getMenuAction().getId() != MenuAction.RUNELITE.getId())
+		{
+			return;
+		}
+
+		final Skill skill = Skill.ofName(Text.removeTags(event.getMenuTarget()));
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerPlugin.java]: Skill.valueOf works, you just need to handle the exception

**Pull request merged**